monai/transforms/mri_utils.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import torch
+
+
+def to_tensor(data):
+    """
+    Convert numpy array to PyTorch tensor. For complex arrays, the real and imaginary parts
+    are stacked along the last dimension.
+    inputs:
+        data (np.array): Input numpy array
+    outputs:
+        torch.Tensor: PyTorch version of data
+    """
+    if np.iscomplexobj(data):
+        data = np.stack((data.real, data.imag), axis=-1)
+    return torch.from_numpy(data)
+
+
+def complex_abs(x):
+    """
+    Compute the absolute value of a complex array.
+    inputs:
+        x (np.array): Input numpy array with 2 channels in the last
+        dimension representing real and imaginary parts.
+    outputs:
+        np.array: Absolute value along the last dimention
+    """
+    assert x.shape[-1] == 2
+    return np.sqrt(x[..., 0] ** 2 + x[..., 1] ** 2)
+
+
+def roll(x, shift, dim):
+    """
+    Similar to np.roll but applies to PyTorch Tensors
+    """
+    if isinstance(shift, (tuple, list)):
+        assert len(shift) == len(dim)
+        for s, d in zip(shift, dim):
+            x = roll(x, s, d)
+        return x
+    shift = shift % x.size(dim)
+    if shift == 0:
+        return x
+    left = x.narrow(dim, 0, x.size(dim) - shift)
+    right = x.narrow(dim, x.size(dim) - shift, shift)
+    return torch.cat((right, left), dim=dim)
+
+
+def fftshift(x, dim=None):
```

Comment:
Would it be better to name the func as `fftshift_t()` ? So users won't get confused when importing it and numpy func at the same time

Reply:
I merged these functions into two functions ifftn and fftn both handling NdarrayOrTensor.